```console
% make t T=TestAccCodePipeline_ K=codepipeline 
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-aws_codepipeline-add_commands 🌿...
TF_ACC=1 go1.26.5 test ./internal/service/codepipeline/... -v -count 1 -parallel 20 -run='TestAccCodePipeline_'  -timeout 360m -vet=off
2026/07/21 17:35:57 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/21 17:35:57 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccCodePipeline_List_basic
=== PAUSE TestAccCodePipeline_List_basic
=== RUN   TestAccCodePipeline_List_includeResource
=== PAUSE TestAccCodePipeline_List_includeResource
=== RUN   TestAccCodePipeline_List_regionOverride
=== PAUSE TestAccCodePipeline_List_regionOverride
=== RUN   TestAccCodePipeline_basic
=== PAUSE TestAccCodePipeline_basic
=== RUN   TestAccCodePipeline_disappears
=== PAUSE TestAccCodePipeline_disappears
=== RUN   TestAccCodePipeline_emptyStageArtifacts
=== PAUSE TestAccCodePipeline_emptyStageArtifacts
=== RUN   TestAccCodePipeline_deployWithServiceRole
=== PAUSE TestAccCodePipeline_deployWithServiceRole
=== RUN   TestAccCodePipeline_tags
=== PAUSE TestAccCodePipeline_tags
=== RUN   TestAccCodePipeline_MultiRegion_basic
=== PAUSE TestAccCodePipeline_MultiRegion_basic
=== RUN   TestAccCodePipeline_MultiRegion_update
=== PAUSE TestAccCodePipeline_MultiRegion_update
=== RUN   TestAccCodePipeline_MultiRegion_convertSingleRegion
=== PAUSE TestAccCodePipeline_MultiRegion_convertSingleRegion
=== RUN   TestAccCodePipeline_withNamespace
=== PAUSE TestAccCodePipeline_withNamespace
=== RUN   TestAccCodePipeline_withGitHubV1SourceAction
    codepipeline_test.go:499: skipping test; environment variable GITHUB_TOKEN must be set
--- SKIP: TestAccCodePipeline_withGitHubV1SourceAction (0.00s)
=== RUN   TestAccCodePipeline_ecr
=== PAUSE TestAccCodePipeline_ecr
=== RUN   TestAccCodePipeline_pipelineType
=== PAUSE TestAccCodePipeline_pipelineType
=== RUN   TestAccCodePipeline_manualApprovalTimeoutInMinutes
=== PAUSE TestAccCodePipeline_manualApprovalTimeoutInMinutes
=== RUN   TestAccCodePipeline_conditions
=== PAUSE TestAccCodePipeline_conditions
=== RUN   TestAccCodePipeline_computeAction
=== PAUSE TestAccCodePipeline_computeAction
=== RUN   TestAccCodePipeline_trigger
=== PAUSE TestAccCodePipeline_trigger
=== CONT  TestAccCodePipeline_List_basic
=== CONT  TestAccCodePipeline_MultiRegion_update
=== CONT  TestAccCodePipeline_manualApprovalTimeoutInMinutes
=== CONT  TestAccCodePipeline_ecr
=== CONT  TestAccCodePipeline_pipelineType
=== CONT  TestAccCodePipeline_withNamespace
=== CONT  TestAccCodePipeline_tags
=== CONT  TestAccCodePipeline_computeAction
=== CONT  TestAccCodePipeline_MultiRegion_basic
=== CONT  TestAccCodePipeline_trigger
=== CONT  TestAccCodePipeline_disappears
=== CONT  TestAccCodePipeline_conditions
=== CONT  TestAccCodePipeline_emptyStageArtifacts
=== CONT  TestAccCodePipeline_deployWithServiceRole
=== CONT  TestAccCodePipeline_MultiRegion_convertSingleRegion
=== CONT  TestAccCodePipeline_basic
=== CONT  TestAccCodePipeline_List_regionOverride
=== CONT  TestAccCodePipeline_List_includeResource
=== NAME  TestAccCodePipeline_emptyStageArtifacts
    codepipeline_test.go:179: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating S3 Bucket (tf-test-pipeline-test-d4lozsz4fd): operation error S3: CreateBucket, https response error StatusCode: 0, RequestID: , HostID: , request send failed, Put "https://tf-test-pipeline-test-d4lozsz4fd.s3.us-west-2.amazonaws.com/": dial tcp: lookup tf-test-pipeline-test-d4lozsz4fd.s3.us-west-2.amazonaws.com: no such host
        
          with aws_s3_bucket.test,
          on terraform_plugin_test.tf line 12, in resource "aws_s3_bucket" "test":
          12: resource "aws_s3_bucket" "test" {
        
--- FAIL: TestAccCodePipeline_emptyStageArtifacts (20.31s)
--- PASS: TestAccCodePipeline_List_regionOverride (46.99s)
--- PASS: TestAccCodePipeline_disappears (58.23s)
--- PASS: TestAccCodePipeline_List_basic (59.31s)
--- PASS: TestAccCodePipeline_withNamespace (60.66s)
--- PASS: TestAccCodePipeline_computeAction (62.43s)
--- PASS: TestAccCodePipeline_MultiRegion_basic (65.12s)
--- PASS: TestAccCodePipeline_List_includeResource (67.98s)
--- PASS: TestAccCodePipeline_ecr (72.70s)
--- PASS: TestAccCodePipeline_deployWithServiceRole (75.84s)
--- PASS: TestAccCodePipeline_basic (95.95s)
--- PASS: TestAccCodePipeline_trigger (96.76s)
--- PASS: TestAccCodePipeline_manualApprovalTimeoutInMinutes (99.06s)
--- PASS: TestAccCodePipeline_MultiRegion_update (100.19s)
--- PASS: TestAccCodePipeline_conditions (103.06s)
--- PASS: TestAccCodePipeline_tags (113.87s)
--- PASS: TestAccCodePipeline_MultiRegion_convertSingleRegion (131.59s)
--- PASS: TestAccCodePipeline_pipelineType (183.52s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/codepipeline       188.547s
FAIL
make: *** [t] Error 1
subhammukhopadhyay@Subhams-MacBook-Pro terraform-provider-aws % make t T=TestAccCodePipeline_emptyStageArtifacts K=codepipeline
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-aws_codepipeline-add_commands 🌿...
TF_ACC=1 go1.26.5 test ./internal/service/codepipeline/... -v -count 1 -parallel 20 -run='TestAccCodePipeline_emptyStageArtifacts'  -timeout 360m -vet=off
2026/07/21 17:52:39 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/21 17:52:39 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccCodePipeline_emptyStageArtifacts
=== PAUSE TestAccCodePipeline_emptyStageArtifacts
=== CONT  TestAccCodePipeline_emptyStageArtifacts
--- PASS: TestAccCodePipeline_emptyStageArtifacts (44.73s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/codepipeline       49.696s
```

--- a/.changelog/42507.txt
+++ b/.changelog/42507.txt
@@ -1,3 +1,7 @@
 ```release-note:enhancement
-resource/aws_codepipeline: Support for `Compute` action
+resource/aws_codepipeline: Added new arguments `commands` and 'output_artifacts_for_compute_action' to support 'Compute' action
+```
+
+```release-note:enhancement
+resource/aws_codepipeline: 'output_artifacts_for_compute_action' and 'output_artifacts' now conflict
 ```

--- a/internal/service/codepipeline/codepipeline_test.go
+++ b/internal/service/codepipeline/codepipeline_test.go
@@ -1366,6 +1366,48 @@ func TestAccCodePipeline_trigger(t *testing.T) {
 	})
 }
 
+func TestAccCodePipeline_computeActionWithOutputAttributes(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := acctest.RandString(t, 10)
+
+	acctest.ParallelTest(ctx, t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			testAccPreCheck(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.CodePipelineServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckPipelineDestroy(ctx, t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCodePipelineConfig_computeActionWithOutputArtifacts(rName),
+				ExpectError: regexache.MustCompile(`stage.\d+.action.\d+: only one of "output_artifacts" or "output_artifacts_for_compute_action" may be set`),
+			},
+		},
+	})
+}
+
+func TestAccCodePipeline_nonComputeActionWithOutputAttributesForComputeAction(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := acctest.RandString(t, 10)
+
+	acctest.ParallelTest(ctx, t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			testAccPreCheck(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.CodePipelineServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckPipelineDestroy(ctx, t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCodePipelineConfig_nonComputeActionWithOutputAttributesForComputeAction(rName),
+				ExpectError: regexache.MustCompile(`stage.\d+.action.\d+: "output_artifacts_for_compute_action" can only be set when category is "Compute"`),
+			},
+		},
+	})
+}
+
 func testAccCheckPipelineExists(ctx context.Context, t *testing.T, n string, v *types.PipelineDeclaration) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
@@ -3718,6 +3760,146 @@ resource "aws_codepipeline" "test" {
     }
   }
 }
+resource "aws_codestarconnections_connection" "test" {
+  name          = %[1]q
+  provider_type = "GitHub"
+}
+`, rName))
+}
+
+func testAccCodePipelineConfig_computeActionWithOutputArtifacts(rName string) string { // nosemgrep:ci.codepipeline-in-func-name
+	return acctest.ConfigCompose(
+		testAccS3DefaultBucket(rName),
+		testAccServiceIAMRole(rName),
+		fmt.Sprintf(`
+resource "aws_codepipeline" "test" {
+  name          = "test-pipeline-%[1]s"
+  role_arn      = aws_iam_role.codepipeline_role.arn
+  pipeline_type = "V2"
+
+  artifact_store {
+    location = aws_s3_bucket.test.bucket
+    type     = "S3"
+
+    encryption_key {
+      id   = "1234"
+      type = "KMS"
+    }
+  }
+
+  stage {
+    name = "Source"
+
+    action {
+      name             = "Source"
+      category         = "Source"
+      owner            = "AWS"
+      provider         = "CodeStarSourceConnection"
+      version          = "1"
+      output_artifacts = ["test"]
+
+      configuration = {
+        ConnectionArn    = aws_codestarconnections_connection.test.arn
+        FullRepositoryId = "lifesum-terraform/test"
+        BranchName       = "main"
+      }
+    }
+  }
+
+  stage {
+    name = "Build"
+
+    action {
+      name            = "Build"
+      category        = "Compute"
+      owner           = "AWS"
+      provider        = "Commands"
+      input_artifacts = ["test"]
+      version         = "1"
+
+      commands = [
+        "echo hello1 > hello1.txt",
+        "echo hello2 > hello2.txt",
+        "echo hello3 > hello3.txt",
+      ]
+
+      configuration = {}
+      output_artifacts = ["test"]
+      output_artifacts_for_compute_action {
+        name  = "ComputeArtifact"
+        files = ["hello1.txt", "hello2.txt", "hello3.txt"]
+      }
+      output_variables = ["AWS_DEFAULT_REGION"]
+    }
+  }
+}
+resource "aws_codestarconnections_connection" "test" {
+  name          = %[1]q
+  provider_type = "GitHub"
+}
+`, rName))
+}
+
+func testAccCodePipelineConfig_nonComputeActionWithOutputAttributesForComputeAction(rName string) string { // nosemgrep:ci.codepipeline-in-func-name
+	return acctest.ConfigCompose(
+		testAccS3DefaultBucket(rName),
+		testAccServiceIAMRole(rName),
+		fmt.Sprintf(`
+resource "aws_codepipeline" "test" {
+  name     = "test-pipeline-%[1]s"
+  role_arn = aws_iam_role.codepipeline_role.arn
+
+  artifact_store {
+    location = aws_s3_bucket.test.bucket
+    type     = "S3"
+
+    encryption_key {
+      id   = "1234"
+      type = "KMS"
+    }
+  }
+
+  stage {
+    name = "Source"
+
+    action {
+      name             = "Source"
+      category         = "Source"
+      owner            = "AWS"
+      provider         = "CodeStarSourceConnection"
+      version          = "1"
+      output_artifacts = ["test"]
+
+      configuration = {
+        ConnectionArn    = aws_codestarconnections_connection.test.arn
+        FullRepositoryId = "lifesum-terraform/test"
+        BranchName       = "main"
+      }
+    }
+  }
+
+  stage {
+    name = "Build"
+
+    action {
+      name            = "Build"
+      category        = "Build"
+      owner           = "AWS"
+      provider        = "CodeBuild"
+      input_artifacts = ["test"]
+      version         = "1"
+
+      configuration = {
+        ProjectName = "test"
+      }
+      output_artifacts_for_compute_action {
+        name  = "ComputeArtifact"
+        files = ["hello1.txt", "hello2.txt", "hello3.txt"]
+      }
+    }
+  }
+}
+
 resource "aws_codestarconnections_connection" "test" {
   name          = %[1]q
   provider_type = "GitHub"

--- a/internal/service/codepipeline/codepipeline_test.go
+++ b/internal/service/codepipeline/codepipeline_test.go
@@ -3823,7 +3823,7 @@ resource "aws_codepipeline" "test" {
         "echo hello3 > hello3.txt",
       ]
 
-      configuration = {}
+      configuration    = {}
       output_artifacts = ["test"]
       output_artifacts_for_compute_action {
         name  = "ComputeArtifact"

